### PR TITLE
Full Illinois Model

### LINKED ISSUE #11: Add multiple years to the model time horizon
This issue may be closed when there is a PR that adds multiple years to the model time horizon. 

Guidance for adding this feature to the model may be found in the [PyPSA documentation](https://pypsa.readthedocs.io/en/latest/examples/multi-investment-optimisation.html).

To start out, there should be at least two years, 2030 and 2050.

Additional considerations
- Each build year in the model time horizon should have specific generators for that year.
- The capital and marginal costs should be updated for each year in the model horizon corresponding to the years in NREL's ATB.

### LINKED ISSUE #21: Create a separate rule for plotting results after solve network
Currently, the `solve_network` rule performs two functions: solving the network AND plotting a dispatch sample.

This issue may be closed when there is a PR that separates this functionality into two unique rules.

### LINKED ISSUE #23: Add ramp rate parameters to technologies
Currently, conventional technologies have infinite ramp rates. This leads to a modeling artifact where nuclear energy demonstrates some amount of load following, which doesn't really happen in practice.

This issue may be closed when there is a PR that add ramp rates to thermal technologies.

### LINKED ISSUE #25: Load growth parameters
Currently, total load demand is taken from historical load data. Instead, we should use historical data for the load _shape_ and renormalize according to projected amount of power demand. 

E.g., 

Total demand = 144e6 MWh

demand = historical demand / sum(historical demand) * total demand

### LINKED ISSUE #26: Add build years to each technology
Currently, the model builds all technologies at once. This should be updated to build 
- [ ] existing technologies (since they're aggregated, the build year should be the average of all existing plants)
- [ ] technology for each year in the model years.

### PR DESCRIPTION
**Pull request by samgdotson**
This PR adds a full model of Illinois in the PyPSA framework using a snakemake workflow.

This pull request:
closes #11 
closes #21 
closes #23
closes #25 
closes #26 

There are still some known issues with the model. In particular,
* projected load growth
* cost assumptions/projections for solar and wind generators
* fuel price assumptions for natural gas generators

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -1,27 +1,53 @@
 state_abbr: 'IL'
 
 geo_res: 'rto'  # accepts: 'rto' or 'county'
-time_res: 4  # hours 
+time_res: 1  # hours 
+load_filter: 60e3  # MW, load above this level will be removed as outliers.
+total_demand: 136e6  # Annual MWh in the first year
+load_growth: 0.00  # % annual growth
+load_share:
```
Comment (lshaver) [inline config.yml:8]: How/where is this used?

File: config.yml [hunk 2]
```diff
@@ -1,27 +1,53 @@
 state_abbr: 'IL'
 
 geo_res: 'rto'  # accepts: 'rto' or 'county'
-time_res: 4  # hours 
+time_res: 1  # hours 
+load_filter: 60e3  # MW, load above this level will be removed as outliers.
+total_demand: 136e6  # Annual MWh in the first year
+load_growth: 0.00  # % annual growth
```
Comment (lshaver) [inline config.yml:7]: When choosing a value for this, may be more instructive to consider the total in the final model year and select a rate that grows to that value.

File: config.yml (hunk 1 above)
Comment (samgdotson) [reply config.yml:8]: this parameter is not used, anywhere. I think I intended to assign a fraction of the total demand to each bus region, but it was unnecessary.

File: config.yml (hunk 2 above)
Comment (samgdotson) [reply config.yml:7]: Instructive in what way?

**Review by lshaver — APPROVED** [review]
Was able to run the model and view results. Future updates should add some instructions to the readme about updating the environment and clearing snakemake results when making changes to the model. There are also some hard coded values in some scripts related to ramp rates, number of reactors, etc, that should probably be moved to the config file in future iterations, for better generalization.

**Pull request merged**